pkg/test/framework/test.go

```diff
@@ -181,6 +185,39 @@ func (t *testImpl) RunParallel(fn func(ctx TestContext)) {
 	t.runInternal(fn, true)
 }
 
+// EchoSet contains a common set of apps
+type EchoSet interface {
+	Deployments() map[string]echo.Instances
+	All() echo.Instances
+}
+
+func (t *testImpl) RunForAll(apps EchoSet, fn func(ctx TestContext, src echo.Instance, dest echo.Instances)) {
+	t.runInternal(func(ctx TestContext) {
+		for srcName, srcs := range apps.Deployments() {
+			srcName, srcs := srcName, srcs
+			ctx.NewSubTest("from " + srcName).Run(func(ctx TestContext) {
+				if len(srcs) == 0 {
+					ctx.Skipf("no instances for source %s", srcName)
```

Comment:
when would this happen? would they just not have it in apps.Deployment if they don't want something?

If it does happen, should we even start a subtest or just silently drop it?

Reply:
https://github.com/istio/istio/blob/2fcdd86f2f7dd21bb9a7091feb5d14733435ef63/pkg/test/framework/components/echo/echoboot/echoboot.go#L78-L80

If the above is triggered, then the following would make `apps.VM` have len 0. 

https://github.com/istio/istio/blob/0b7570523ab2571dc35e934642922b774a4b789b/tests/integration/pilot/common/apps.go#L237

I think it's preferable to be noisy about anything specified outside the test that isn't actually being run. If the VM wasn't event attempted to be deployed and somehow not in the `Deployments()` map we wouldn't get this warning.